File: QViz.NET/Periscope.QViz/JSON/Resource.cs
using System;
using System.Collections.Generic;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for Resource Details
	/// </summary>
	public class Resource
	{
		/// <summary>
		/// Identifier of the Resource
		/// </summary>
		public string resourceId;
		
		/// <summary>
		/// Name of the Resource
		/// </summary>
		public string name;
		
		/// <summary>
		/// Type of the Resource
		/// </summary>
		public string type;
		
		/// <summary>
		/// Is this Resource Active?
		/// </summary>
		public bool isActive;
		
		/// <summary>
		/// Date and Time on which this Model was Created
		/// </summary>
		public DateTime createdOn;
		
		/// <summary>
		/// Date and Time on which this Model was last Updated
		/// </summary>
		public DateTime updatedOn;
		
		/// <summary>
		/// List of User Role Permission Objects
		/// </summary>
		public IList<RolePermission> rolePermissions;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public Resource()
		{
			this.resourceId = "";
			this.name = "";
			this.type = "";
			this.isActive = false;
			this.createdOn = DateTime.Now;
			this.updatedOn = DateTime.Now;
			this.rolePermissions = new List<RolePermission>();
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/Resource.java
package com.periscope.qviz.json;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * JSON Model for Resource Details
 */
public class Resource {

	/**
	 * Identifier of the Resource
	 */
	public String resourceId;

	/**
	 * Name of the Resource
	 */
	public String name;

	/**
	 * Type of the Resource
	 */
	public String type;

	/**
	 * Is this Resource Active?
	 */
	public Boolean isActive;

	/**
	 * Date and Time on which this Model was Created
	 */
	public Date createdOn;

	/**
	 * Date and Time on which this Model was last Updated
	 */
	public Date updatedOn;

	/**
	 * List of User Role Permission Objects
	 */
	public List<RolePermission> rolePermissions;

	/**
	 * Default Constructor
	 */
	public Resource() {
		this.resourceId = "";
		this.name = "";
		this.type = "";
		this.isActive = false;
		this.createdOn = null;
		this.updatedOn = null;
		this.rolePermissions = new ArrayList<>();
	}

}
